### PR TITLE
client: allows workflow creation from json

### PR DESCRIPTION
**Pull request by roksys**
* Adding new feature which allows to create a workflow form
  json. Connects #228

Signed-off-by: Rokas Maciulaitis <rokas.maciulaitis@cern.ch>

### CODE REVIEW

File: reana_client/api/client.py [hunk 1]
```diff
@@ -131,8 +131,85 @@ def create_workflow(reana_specification, name, access_token):
...
+         >>>    workflow_json=workflow_json,
+         >>>    name='workflow_name.1',
+         >>>    access_token='access_token',
+         >>>    parameters={'files': ['file.txt'],
+                            'parameters': {'key': 'value'},
+         >>>    workflow_engine = 'serial')
+
+    """
+    if is_uuid_v4(name):
+        raise Exception('Workflow name cannot be a valid UUIDv4')
```
Comment (dinosk) [inline reana_client/api/client.py]: `ValueError`?

File: reana_client/api/client.py [hunk 2]
```diff
@@ -131,8 +131,85 @@ def create_workflow(reana_specification, name, access_token):
...
+    if not access_token:
+        raise Exception(ERROR_MESSAGES['missing_access_token'])
+    if os.environ.get('REANA_SERVER_URL') is None:
+        raise Exception('Environment variable REANA_SERVER_URL is not set')
+    workflow_engine = workflow_engine.lower()
+    workflow_engines = ['yadage', 'cwl', 'serial']
```
Comment (dinosk) [inline reana_client/api/client.py]: Could be a config var like `WORKFLOW_ENGINES`

File: reana_client/api/client.py [hunk 3]
```diff
@@ -131,8 +131,84 @@ def create_workflow(reana_specification, name, access_token):
...
+    :param workflow_json: workflow specification in json format.
+    :param name: name or UUID of the workflow to be started.
+    :param access_token: Access token of the current user.
+    :param parameters: workflow input parameters dictionary.
+    :param workflow_engine: one of the workflow engines (yadage, serial, cwl)
+    :param outputs: dictionary with expected workflow outputs.
+
+    :Example:
+
+         >>> create_workflow_from_json(
```
Comment (diegodelemos) [inline reana_client/api/client.py]: 👍 ! I am not getting this displayed when building the docs though.

Comment (roksys) [reply reana_client/api/client.py]: screenshot 2019-01-24 at 13 37 32
Does it look like this?

Comment (diegodelemos) [reply reana_client/api/client.py]: I am doing: 
```console
$ ~/reana-client git:(8e1aa70) $ python setup.py build_sphinx
running build_sphinx
Running Sphinx v1.8.2
making output directory...
building [mo]: all of 0 po files
building [html]: all source files
updating environment: 11 added, 0 changed, 0 removed
reading sources... [100%] userguide
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] userguide
generating indices... genindex
writing additional pages... search
copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded.

The HTML pages are in docs/_build/html.
```
image

**Comment by diegodelemos** [comment]
Looks great, let's see if we have all docs correctly displayed and with the small comments should be ready 🚀

**Pull request merged**